src/backend/distributed/README.md

```diff
@@ -1280,7 +1282,7 @@ Correlated or LATERAL subqueries have special behavior in Citus. They can often
  For more information on the code, check the following functions:
  `DeferErrorIfCannotPushdownSubquery()` ->
  `ContainsReferencesToOuterQuery()` ->
```

Comment:
I think this suggestion is wrong, I meant `DeferErrorIfSubqueryRequiresMerge`. So, LATERAL queries are different/unique, even if the subquery requires `merge step` such as a `LIMIT`, if the correlation is on the distribution column, we can push it down. See https://github.com/citusdata/citus/pull/4385

And, I wanted to point that out here. Maybe add a link to this PR to clarify?

`DeferredErrorIfUnsupportedLateralSubquery` is interesting by its own, I'm fine to add it, but lets keep `DeferErrorIfSubqueryRequiresMerge` as well